File: 04.jni/airplay/app/src/main/java/org/huihui/airplay/MainActivity.java
package org.huihui.airplay;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tv = (TextView) findViewById(R.id.sample_text);
        MDNS.register(6000, "huihui");
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    DatagramSocket datagramSocket = new DatagramSocket(80);
                    InetAddress address = InetAddress.getByName("192.168.199.222");
                    DatagramPacket p = new DatagramPacket(new byte[]{1, 2, 3}, 3,address,80);
                    while (true) {
                        datagramSocket.send(p);
                        Thread.sleep(1000);
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }).start();
        new AirPlay().init(80,"111");
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
}

File: 04.jni/airplay/app/src/main/java/org/huihui/airplay/MainActivity.kt
package org.huihui.airplay

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.InetAddress

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        MDNS.register(6000, "huihui")
        Thread(Runnable {
            try {
                val datagramSocket = DatagramSocket(80)
                val address = InetAddress.getByName("192.168.199.222")
                val p = DatagramPacket(byteArrayOf(1, 2, 3), 3, address, 80)
                while (true) {
                    datagramSocket.send(p)
                    Thread.sleep(1000)
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }).start()
        val airPlay = AirPlay()
        airPlay.init(80, "111")
        airPlay.start()
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
}
